src/conspiracies/prompt_relation_extraction/template_class.py

```diff
@@ -59,6 +72,15 @@ def set_examples(self, examples: Tuple[List[Doc], List[List[SpanTriplet]]]):
 
 
 class PromptTemplate1(PromptTemplate):
+    def __init__(
+        self,
+        task_description: Optional[str] = None,
+        examples: Optional[Tuple[List[Doc], List[List[SpanTriplet]]]] = None,
+    ):
+        super().__init__(task_description, examples)
+        if not task_description:
+            self.task_description = self.default_task_descriptions["parentheses"]
+
```

Comment:
Would probably just add the description to the task to avoid coupling between the classes). I.e. if I want to change the default for template1 it will now also change template 2.

```suggestion
    def __init__(
        self,
        task_description: Optional[str] = None,
        examples: Optional[Tuple[List[Doc], List[List[SpanTriplet]]]] = None,
    ):
        super().__init__(task_description, examples)
        if not task_description:
            self.task_description = """Extract semantic triplets from the following tweet. 
 The semantic triplets should be on the form (Subject - Verb Phrase - Object), where the verb phrase includes all particles and modifyers. 
 There should always be exactly three elements in a triplet, no more no less. 
 They should be presented with each element in parentheses as shown below:"""

```

Reply:
that will naturally lead to the string appearing twice, but I think that is desirable in this case.

Reply:
Yes that it why it wrote like this at least, but it does make sense to make them independent as you say. Will fix :))